newdle/client/src/components/ErrorMessage.js

```diff
@@ -0,0 +1,31 @@
+import React, {useEffect} from 'react';
+import {useDispatch} from 'react-redux';
+import PropTypes from 'prop-types';
+import {useHistory} from 'react-router';
+import {Container, Message} from 'semantic-ui-react';
+import {clearError} from '../actions';
+
+export default function ErrorMessage({error}) {
+  const history = useHistory();
+  const dispatch = useDispatch();
+
+  const unregister = history.listen(() => {
+    dispatch(clearError());
+  });
+
+  useEffect(() => {
+    return () => {
+      unregister();
+    };
+  }, [unregister]);
```

Comment:
I'm not sure I understand what this is supposed to do. Maybe we could add some comments?

Reply:
This is responsible for removing the callback registered with `listen` once the component gets unmounted.